build-logic/build.gradle.kts

```diff
@@ -17,6 +24,39 @@ kotlin {
             .get()
             .toInt(),
     )
+
+    compilerOptions {
+        // Match language version enforced by current Gradle version https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+        @Suppress("DEPRECATION")
+        apiVersion.set(KotlinVersion.KOTLIN_1_8)
```

Comment:
To futureproof the build, I added extra config that makes sure it's not going to be possible to use kotlin code not available when running the build. 

With Gradle [9.0](https://docs.gradle.org/9.0.0-rc-1/userguide/compatibility.html#kotlin) this should be bumped to `2.1`, but there will be nothing wrong if it lags behind a bit 🤷